### PR TITLE
add transformation to simplex

### PR DESCRIPTION
**Pull request by scheidan**
I've added a `UnitSimplex` transformation (see issue #43 ).

Not sure if I have to do somethings with the docs

### CODE REVIEW

**Comment by tpapp** [comment]
Thank you, I will review this soon. I will add the docs too.

**Comment by tpapp** [comment]
Very nicely written and clean code, thanks! I just have a couple of minor comments.

File: test/runtests.jl [hunk 1]
```diff
@@ -103,6 +103,27 @@ end
...
+        @test_throws ArgumentError S(x)
+        @test_throws ArgumentError transform(S, x)
+        @test_throws ArgumentError transform_and_logjac(S, x)
+    end
+
+    @testset "consistency checks" begin
+        for K in 1:10
+            t = UnitSimplex(K)
+            @test dimension(t) == K - 1
+            if K > 1
```
[the remark this answers is not in the document]
Comment (scheidan) [reply test/runtests.jl]: Good idea! I added a test for that

**Comment by tpapp** [comment]
Thanks for the very nice contribution! Will do the docs and then tag a release.

**Comment by scheidan** [comment]
Thanks fo the fast merge! 
(I think this also closes #43)

**Pull request merged**